libkbfs/chat_rpc.go

```diff
@@ -296,3 +309,180 @@ func (c *ChatRPC) ReadChannel(
 	}
 	return messages, nextPage, nil
 }
+
+// RegisterForMessages implements the Chat interface.
+func (c *ChatRPC) RegisterForMessages(
+	convID chat1.ConversationID, cb ChatChannelNewMessageCB) {
+	str := convID.String()
+	c.convLock.Lock()
+	defer c.convLock.Unlock()
+	c.convCBs[str] = append(c.convCBs[str], cb)
+}
+
+// We only register for the kbfs-edits type of notification in
+// keybase_daemon_rpc, so all the other methods below besides
+// `NewChatKBFSFileEditActivity` should never be called.
+var _ chat1.NotifyChatInterface = (*ChatRPC)(nil)
+
+// NewChatActivity implements the chat1.NotifyChatInterface for
+// ChatRPC.
+func (c *ChatRPC) NewChatActivity(
+	_ context.Context, _ chat1.NewChatActivityArg) error {
+	return nil
+}
+
+// NewChatKBFSFileEditActivity implements the
+// chat1.NotifyChatInterface for ChatRPC.
+func (c *ChatRPC) NewChatKBFSFileEditActivity(
+	ctx context.Context, arg chat1.NewChatKBFSFileEditActivityArg) error {
+	activityType, err := arg.Activity.ActivityType()
+	if err != nil {
+		return err
+	}
+	switch activityType {
+	case chat1.ChatActivityType_NEW_CONVERSATION:
+		// If we learn about a new conversation for a given TLF,
+		// attempt to route it to the TLF.
+		info := arg.Activity.NewConversation()
+		tlfType := tlf.Private
+		if info.Conv.Visibility == keybase1.TLFVisibility_PUBLIC {
+			tlfType = tlf.Public
```

Comment:
Does this mean that notifications for public TLFs will be public?

I guess I only subscribe to them if I have a public TLF as a favorite, correct?

Reply:
Yeah, notifications are public, but as chris said today in the call, we'll only include your own notifications for your own public folder in the overall list.  The RPC inbox call doesn't even return public chats, we're going to have to add that in manually at a higher layer.